3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/remove_hard_coded_jurisdiction_depth.psql

```diff
@@ -129,7 +125,17 @@ SELECT DISTINCT ON (locations.id, events_query.task_id)
     COALESCE(events_query.notsprayed_reasons, ARRAY[]::varchar[]) AS notsprayed_reasons,
     COALESCE(events_query.duplicate, FALSE) AS duplicate,
     COALESCE(events_query.sprayed_duplicate, FALSE) AS sprayed_duplicate
-FROM locations
+FROM irs_plans
+LEFT JOIN plan_jurisdiction
+    ON plan_jurisdiction.plan_id = irs_plans.plan_id
+LEFT JOIN jurisdictions_tree AS jurisdictions_tree
+    ON plan_jurisdiction.jurisdiction_id = jurisdictions_tree.jurisdiction_id
+LEFT JOIN locations
+    ON (
+        plan_jurisdiction.jurisdiction_id = locations.jurisdiction_id OR
+        -- this takes care of the possibility of higher level jurisdictions in the plan
+        locations.jurisdiction_id = ANY(jurisdictions_tree.jurisdiction_path)
+    )
```

Comment:
We are selecting from irs_plans and then joining locations using the plan's jurisdictions.  This should get us all the locations that are relevant to a plan.